applications/FemToDemApplication/custom_processes/compute_normalized_free_energy_on_nodes_process.cpp

```diff
@@ -0,0 +1,335 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics FemDem Application
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo Velazquez
+//
+
+#include "custom_processes/compute_normalized_free_energy_on_nodes_process.h"
+
+namespace Kratos
+{
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+// Constructor
+ComputeNormalizedFreeEnergyOnNodesProcess::ComputeNormalizedFreeEnergyOnNodesProcess(
+    ModelPart &r_model_part, 
+    int Dimension, 
+    Parameters ThisParameters) : mrModelPart(r_model_part)
+{
+    mNNodes = mrModelPart.NumberOfNodes();
+    mDimension = Dimension;
+
+    Parameters default_parameters = Parameters(R"(
+    {
+        "normalized_free_energy" : false,
+        "correct_with_displacements" : true
+    })" );
+    ThisParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+    mComputeNormalizedFreeEnergy = ThisParameters["normalized_free_energy"].GetBool();
+    mCorrectWithDisplacements = ThisParameters["correct_with_displacements"].GetBool();
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ComputeNormalizedFreeEnergyOnNodesProcess::Execute()
+{
+    int max_id;
+    this->ObtainMaximumNodeId(max_id);
+    NodeNormalizedFreeEnergy *NodeNormalizedFreeEnergyVector = new NodeNormalizedFreeEnergy[max_id];
+    this->NormalizedFreeEnergyExtrapolation(NodeNormalizedFreeEnergyVector);
+    delete[] NodeNormalizedFreeEnergyVector;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ComputeNormalizedFreeEnergyOnNodesProcess::NormalizedFreeEnergyExtrapolation(
+    NodeNormalizedFreeEnergy *pNodeNormalizedFreeEnergyVector
+    )
+{
+    Vector gauss_point_stress;
+    double gauss_point_damage;
+    // Loop over elements to extrapolate the stress to the nodes
+    for (auto it_elem = mrModelPart.Elements().ptr_begin(); it_elem != mrModelPart.Elements().ptr_end(); ++it_elem) {
+        auto& r_geometry = (*it_elem)->GetGeometry();
+        auto& r_mat_properties = (*it_elem)->GetProperties();
+
+        const Vector& r_strain_vector = (*it_elem)->GetValue(STRAIN_VECTOR);
```

Comment:
You don't need the *
```suggestion
        const Vector& r_strain_vector = it_elem->GetValue(STRAIN_VECTOR);
```